src/core/net/srp_server.cpp

```diff
@@ -1199,6 +1201,68 @@ Error Server::VerifySignature(const Dns::Ecdsa256KeyRecord &aKeyRecord,
     return error;
 }
 
+Error Server::ValidateServiceSubTypes(Host &aHost, const MessageMetadata &aMetadata)
+{
+    Error error = kErrorNone;
+    Host *existingHost;
+
+    // Verify that there is a matching base type service for all
+    // sub-type services in `aHost` (which is from the received
```

Comment:
nit: the RFCs use "subtype" rather than "sub-type"
```suggestion
    // subtype services in `aHost` (which is from the received
```

Reply:
Really like that you did notice it .... 😃 
I also did notice that when adding support for subtype originally and spent sometime thinking of options and decided to use "sub-type" in documentation.

I noticed that within OT code we seem to use `"sub-type"` in documentations. In `Message` class
```
    /**
     * This enumeration represents the message sub-type.
     *
     */
    enum SubType : uint8_t
```

and in `mac_frame`
```c++
    /**
     * This method returns the Vendor IE sub-type.
     *
     * @returns The Vendor IE sub-type.
     *
     */
    uint8_t GetSubType(void) const { return mSubType; }
```
and also in `dtls.hpp`

So I followed the same pattern in SRP/DNS to remain consistent within OT.

Some other observations:
- We do use "sub-" in some other places: "sub-mac", "sub-domain", "sub-TLV", etc.
- The other aspect I considered is the method/variable names. We use `SubType` vs `Subtype` (so we kind of treat it as two nouns), so having the dash "-" somewhat mirrors this naming model.

Reply:
Yeah, I was simply referring to https://datatracker.ietf.org/doc/html/rfc6763#section-7.1, which uses `subtype`.